public/app/features/dashboard-scene/utils/dashboardSessionState.ts

```diff
@@ -18,9 +18,9 @@ export function restoreDashboardStateFromLocalStorage(dashboard: DashboardScene)
       if (!currentQueryParams.has(key)) {
         currentQueryParams.append(key, value);
       } else {
-        if (!currentQueryParams.getAll(key).includes(value)) {
-          currentQueryParams.append(key, value);
-        }
+        // if (!currentQueryParams.getAll(key).includes(value)) {
+        //   currentQueryParams.append(key, value);
+        // }
```

Comment:
```suggestion
```

Reply:
@dprokop am I right in removing this? I feel like it a data link or other link on a page includes a variable or or time range URL key that should override any saved (preserved state). 

Meaning if the URL currently has ´?var-app=A&var-app=B`

and someone has defined a data link or manual link to another dashboard with `?var-app=E` , the final state when opening that dashboard should just be with value `var-app=E` not any of the preserved values. 

That is why I removed that if statement as it will mix values for a key present in the preserved and current query params, I think if a key is present in the URL then that should override any "preservation" state

Reply:
You are right @torkel - if the link contains query param, then that param should take precedence over the persisted one as it's explicit.